scripts/list_shields.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
```

Comment:
Non blocking comment.

The long term goal is to get rid of the `list_boards.py` and unify this under [list_hardware.py](https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/list_hardware.py) because listing of boards now requires sourcing of the SoC for the CPU cluster knowledge (and in future maybe SiP in addition).

Reason for keeping the `list_boards` short term is that we still support HWMv1.

So although `list_shields.py` is small, I think it make sense if this can be done as part of list_hardware.

Of course it can also be reasoned that because shields has not been touched as part of the new hw model and therefore is still looked up using `Kconfig.shield` + `<name>.overlay`  then keeping it independent from list_hardware makes sense.

Reply:
Yeah, I was a bit confused as to the purpose of the new `list_hardware.py` script. I initially tried integrating shield support there, but it seemed like it was more geared towards SoCs and architectures, whereas `list_boards.py` seems closed to what I wanted to add.

I'd prefer to leave `list_shields.py`.